Danki2/Assets/Scripts/AI/Planner.cs

```diff
@@ -0,0 +1,4 @@
+﻿public interface Planner<T> where T : Actor
+{
+    Agenda Plan(T actor, Agenda previousAgenda);
```

Comment:
Having an object rather than a lambda function for the plan allows it to track state which is really important - I think we can get away with tracking state in the plan and not in the behaviours. That would make the concerns of each type a bit clearer too. What do you reckon? If we go that way, behaviours can be static and we can neatly collect them together in static classes.

Reply:
I think you're probably right about just tracking state in the plan being nice. I'd like to keep the behaviours as objects as part of this though as I think it'll important for the way I want to try and use attributes. If this AI component plan doesn't really work though but we still want to bring this stuff into master (it's only going into the feature branch at the moment) I think we should do what you're saying so we can have the static behaviours

Reply:
Ok that makes sense.